### PR TITLE
Scan temperature sensors

### PR DESCRIPTION
**Pull request by mongrelx**
Scans temperature sensors and print values .. didn't write any config file handler to store findings yet.

### CODE REVIEW

**Comment by Chris911** [comment]
I'll test this in the coming days when I have some time but meanwhile can you cleanup the `smc.rb` file? There seems to be a lot of unneeded new lines.

**Comment by mongrelx** [comment]
Cleaned code ... note that this only scans uppercase sensors .. some models seems to have mixed case sensors

**Comment by mongrelx** [comment]
Added mixed case scanning ..

**Comment by mongrelx** [comment]
Did you ever checked this ?

**Comment by Chris911** [comment]
Hey. I've been on vacation without a computer for a few months now. Going back home in a couple weeks and I'll check all the pull requests at that time. 

Thanks for your contribution.

File: lib/iStats/settings.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module IStats
+  class Settings
+    require 'parseconfig'
```
Comment (Chris911) [inline lib/iStats/settings.rb]: Package needs to be added to the gemspec

File: lib/iStats/cpu.rb [hunk 2]
```diff
@@ -15,7 +14,7 @@ def delegate(stat)
         when 'temp', 'temperature'
           cpu_temperature
         else
-          Command.help "Unknown stat for CPU: #{stat}"
+        Command.help "Unknown stat for CPU: #{stat}"
```
Comment (Chris911) [inline lib/iStats/cpu.rb]: Why did you remove the spacing here?

File: lib/iStats/cpu.rb [hunk 3]
```diff
@@ -27,10 +26,15 @@ def all
 
       # Print CPU temperature with sparkline
       #
-      def cpu_temperature
-        t = get_cpu_temp
-        thresholds = [50, 68, 80, 90]
-        puts "CPU temp: #{t}#{Symbols.degree}C  " + Printer.gen_sparkline(t, thresholds)
+      def cpu_temperature
```
Comment (Chris911) [inline lib/iStats/cpu.rb]: I'd rather keep this function to only print temperature from 1 sensor. Perhaps it could be made configurable with the settings you added if the default doesn't work. The flow would be something like:
1. Scan (get list of sensors)
2. `istats set cpu [sensor_key]`

This way the output would still be nice and sweet while allowing custom settings.

File: lib/iStats/settings.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          $config = ParseConfig.new(@configDir+@configFile)
+
+        else
+          puts "No config file #{@configDir}#{@configFile} found"
+          if !File.exists?(@configDir)
+            Dir.mkdir( @configDir)
+          end
+          file=File.open(@configDir+@configFile,"w+")
+          file.close
+          puts "Running scan"
```
Comment (Chris911) [inline lib/iStats/settings.rb]: Loading the settings shouldn't automatically start a scan. Would rather make this explicit with the `scan` option.

**Comment by Chris911** [comment]
This is good work but I'd rather not display all the sensors by default and have the scan run automatically when loading settings (if they are inexistent). If you're up for making other changes let me know and if not I'll merge this and continue working on this.

Thanks for your contribution and sorry it took so long to review.

File: lib/iStats/settings.rb (hunk 4 above)
Comment (mongrelx) [inline lib/iStats/settings.rb]: It now only scans when no config file exists .. i´ll change it to explicit

File: lib/iStats/cpu.rb (hunk 3 above)
Comment (mongrelx) [inline lib/iStats/cpu.rb]: It could create config file with "enabled=0" when key is  not on default list?

Comment (Chris911) [inline lib/iStats/cpu.rb]: Don't think there's a need for a default config. The default sensor for CPU is hardcoded in code (current version). If someone runs the command and there's no information for CPU temp, we can display a message and suggests the user runs a scan and sets the sensor key. It's a bit more completed but for most people the default works so I don't want to make it hard for them. So the flow would become: 
1. Print CPU stats
2. No information found
3. Display help message that suggests running a scan and setting the sensor key in config.
4. User runs scan
5. User sets sensor key using a newly added command.

Comment (mongrelx) [inline lib/iStats/cpu.rb]: All right , i’ll make as you wish .. i’ve long history in linux development , so more bells and whistles ;) 




Sent with Unibox

**Comment by Chris911** [comment]
Busy with work this week but I'll take a look soon.

**Comment by Chris911** [comment]
This is looking good @mongrelx. Merging now, will do some modifications and then release a new version probably today. 

Thanks again for your contribution.
